integration-tests/chopsticks/src/tests/xcm/limitedReseveTransferHydraDxSpiritnet.test.ts

```diff
@@ -0,0 +1,66 @@
+import { test } from 'vitest'
+import { sendTransaction, withExpect } from '@acala-network/chopsticks-testing'
+import { u8aToHex } from '@polkadot/util'
+import { decodeAddress } from '@polkadot/util-crypto'
+
+import * as HydraDxConfig from '../../network/hydraDx.js'
+import * as SpiritnetConfig from '../../network/spiritnet.js'
+import { KILT, initialBalanceKILT, keysAlice, keysBob } from '../../utils.js'
+import { getFreeBalanceHydraDxKilt, getFreeBalanceSpiritnet, hydradxContext, spiritnetContext } from '../index.js'
+
+test('Limited Reserve Transfers from HydraDx Account Bob -> Spiritnet', async ({ expect }) => {
+	const { checkEvents, checkSystemEvents } = withExpect(expect)
+
+	// Create some new blocks to have consistent snapshots
+	await new Promise((r) => setTimeout(r, 50))
+	await spiritnetContext.dev.newBlock()
+	await hydradxContext.dev.newBlock()
+
+	// Give the sovereign account of HydraDx some kilt coins.
+	await spiritnetContext.dev.setStorage(
+		SpiritnetConfig.defaultStorage(u8aToHex(decodeAddress(SpiritnetConfig.hydraDxSovereignAccount)))
+	)
+	await hydradxContext.dev.setStorage(HydraDxConfig.defaultStorage(keysBob.address))
+
+	// check initial balance of alice
+	const aliceBalanceBeforeTx = await getFreeBalanceSpiritnet(keysAlice.address)
+	expect(aliceBalanceBeforeTx).eq(BigInt(0))
```

Comment:
Also here, can we test the same things we test after the operation is executed?

Reply:
For Alice, this is a bit tricky.

1 KILT is moved from the sovereign account to the holding register. The fees for the transaction are paid with that, which finally results in Alice receiving slightly less than the transmitted funds.

Here is the error:

```
AssertionError: expected 99999999999971175n to equal 100000000000000000n

- Expected
+ Received

- 100000000000000000n
+ 99999999999971175n
```

So, do you think rounding is fine? The fees can be slightly different each time, right?

Currently this is checked by checking the Endowed event.

Reply:
I don't think the fees can be different, so it's fine to compare it to the value here and explaining why that's the case, with a comment.

Reply:
[95256c5](https://github.com/KILTprotocol/kilt-node/pull/614/commits/95256c55d9d16d4051d54939d027a7bdecf2120f)